Makefile

```diff
@@ -308,6 +308,7 @@ create-management-cluster: $(KUSTOMIZE) $(ENVSUBST) $(KUBECTL) $(KIND) ## Create
 	./hack/install-cert-manager.sh
 
 	# Create customized cloud provider configs
+	AZURE_CLIENT_ID_USER_ASSIGNED_IDENTITY="$${AZURE_CLIENT_ID_USER_ASSIGNED_IDENTITY:-$$(cat $(AZURE_IDENTITY_ID_FILEPATH))}" \
```

Comment:
Just for my understanding, how is `$$` better?

Reply:
I honestly don't fully understand exactly when `make` evaluates all these variables since many get set during `make kind-create` in between `make create-management-cluster` and `make create-workload-cluster`, but `$$` escapes to one literal `$` to force the shell to evaluate it just-in-time at the moment this command runs instead of `make` itself evaluating this (maybe?) before `make kind-create`.

Reply:
Good breadcrumb trail.